### PR TITLE
Add PowerShellGet 3.0.0-beta8 and CompatPowerShellGet 0.0.1 to csproj

### PR DESCRIPTION
**Pull request by alerickson**
# PR Summary

- Update dependency on PowerShellGet to 3.0.0-beta8
- Remove dependency on PackageManagement
- Add dependency on PowerShellGetCompat 0.0.1

## PR Context

Switching from use of PowerShellGet 2.x (and by extension, PackageManagement) to PowerShellGet 3.x+ and PowerShellGetCompat (a compatibility module to provide a best attempt mapping of v2 to v3 cmdlets). 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [x] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [x] Experimental feature name(s): PowerShellGet
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Comment by iSazonov** [comment]
Convert to Draft until CIs will be fixed.

**Review by adityapatwardhan — APPROVED** [review]

File: test/powershell/Modules/PSDesiredStateConfiguration/configuration.Tests.ps1 [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
 Describe "DSC MOF Compilation" -tags "CI" {
+
+<# PowerShellGet does not yet support DSCResources #>
```
Comment (adityapatwardhan) [inline test/powershell/Modules/PSDesiredStateConfiguration/configuration.Tests.ps1:5]: If there is an issue to track this. Please add a link.

File: test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1 [hunk 2]
```diff
@@ -128,16 +125,16 @@ Describe "PowerShellGet - Module tests" -tags "Feature" {
...
         $installedModuleInfo.Name | Should -Be $ContosoServer
-        $installedModuleInfo.InstalledLocation.StartsWith($script:MyDocumentsModulesPath, [System.StringComparison]::OrdinalIgnoreCase) | Should -BeTrue
+        # Not implemented yet
+        #$installedModuleInfo.InstalledLocation.StartsWith($script:MyDocumentsModulesPath, [System.StringComparison]::OrdinalIgnoreCase) | Should -BeTrue
 
-        $module = Get-Module $ContosoServer -ListAvailable
+        $module = Get-InstalledModule $ContosoServer
         $module.Name | Should -Be $ContosoServer
-        $module.ModuleBase | Should -Be $installedModuleInfo.InstalledLocation
+        #$module.ModuleBase | Should -Be $installedModuleInfo.InstalledLocation
```
Comment (adityapatwardhan) [inline test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1:137]: This seems the most important part of the test as per the title.

File: test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1 [hunk 3]
```diff
@@ -159,16 +156,17 @@ Describe "PowerShellGet - Module tests (Admin)" -Tags @('Feature', 'RequireAdmin
...
 
     It "Should install a module correctly to the required location with AllUsers scope" {
-        Install-Module -Name $ContosoServer -Repository $RepositoryName -Scope AllUsers
+        Install-Module -Name $ContosoServer -Repository $RepositoryName -Scope AllUsers -AllowClobber
         $installedModuleInfo = Get-InstalledModule -Name $ContosoServer
 
         $installedModuleInfo | Should -Not -BeNullOrEmpty
         $installedModuleInfo.Name | Should -Be $ContosoServer
-        $installedModuleInfo.InstalledLocation.StartsWith($script:programFilesModulesPath, [System.StringComparison]::OrdinalIgnoreCase) | Should -BeTrue
+        # Not implemented yet
```
Comment (adityapatwardhan) [inline test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1:164]: Please add a link to an issue tracking this.

File: test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1 [hunk 4]
```diff
@@ -228,24 +226,26 @@ Describe "PowerShellGet - Script tests (Admin)" -Tags @('Feature', 'RequireAdmin
...
-        $installedScriptInfo.InstalledLocation.StartsWith($script:ProgramFilesScriptsPath, [System.StringComparison]::OrdinalIgnoreCase) | Should -BeTrue
+        #$installedScriptInfo.InstalledLocation.StartsWith($script:ProgramFilesScriptsPath, [System.StringComparison]::OrdinalIgnoreCase) | Should -BeTrue
     }
 
     AfterAll {
         Remove-InstalledScripts
     }
 }
 
+<# Currently Not Implemented
```
Comment (adityapatwardhan) [inline test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1:242]: Same as above

File: test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1 [hunk 5]
```diff
@@ -125,21 +124,15 @@ $testCases = @{
...
 
-    "PackageManagement" = @{
-        HelpFiles            = "Microsoft.PowerShell.PackageManagement.dll-help.xml"
-        HelpInfoFiles        = "PackageManagement_4ae9fd46-338a-459c-8186-07f910774cb8_HelpInfo.xml"
-        CompressedFiles      = "PackageManagement_4ae9fd46-338a-459c-8186-07f910774cb8_en-US_helpcontent$extension"
-        HelpInstallationPath = "$PSHOME\Modules\PackageManagement\en-US"
-        HelpInstallationPathHome = "$userHelpRoot\PackageManagement\en-US"
-    }
-
+<# Need to update PowerShellGet information
```
Comment (adityapatwardhan) [inline test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1:127]: File an issue in this repo to track this and add a link.

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: test/powershell/Modules/PSDesiredStateConfiguration/configuration.Tests.ps1 (hunk 1 above)
Comment (alerickson) [reply test/powershell/Modules/PSDesiredStateConfiguration/configuration.Tests.ps1:5]: https://github.com/PowerShell/PowerShellGet/issues/211

File: test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1 (hunk 2 above)
Comment (alerickson) [reply test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1:137]: The object doesn't return this information yet, but I'll include returned object formatting in the next release.  I just commented it out.   Added issue here: https://github.com/PowerShell/PowerShellGet/issues/212

File: test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1 (hunk 3 above)
Comment (alerickson) [reply test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1:164]: https://github.com/PowerShell/PowerShellGet/issues/212

File: test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1 (hunk 4 above)
Comment (alerickson) [reply test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1:242]: https://github.com/PowerShell/PowerShellGet/issues/212

Comment (alerickson) [reply test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1:242]: https://github.com/PowerShell/PowerShellGet/issues/212

File: test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1 (hunk 5 above)
Comment (alerickson) [reply test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1:127]: https://github.com/PowerShell/PowerShell/issues/13332

**Comment by adityapatwardhan** [comment]
@alerickson The PR title says beta9 and code say beta8. Also update the compat module name in title

**Pull request closed without merging**